packages/doc/src/components/CodeBlock/CodeBlock.jsx

```diff
@@ -1,353 +1,97 @@
-import React, { useState } from 'react';
-import styled from 'styled-components';
-import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
-import { MDXContext } from '@mdx-js/react';
-import { node, string } from 'prop-types';
-import Highlight, { defaultProps } from 'prism-react-renderer';
-import { hexToRgb } from '@gympass/yoga-common';
-import * as YogaComponents from '@gympass/yoga';
-import * as YogaIcons from '@gympass/yoga-icons';
-import githubTheme from 'prism-react-renderer/themes/github';
-
-import { CodeSandboxButton } from 'components';
-import CodeIcon from 'images/code.svg';
-import MoonVector from 'images/moon.svg';
-
-const defaultPropsWithTheme = {
-  ...defaultProps,
-  theme: githubTheme,
-};
-
-const StyledLiveError = styled(LiveError)`
-  background-color: #fff0f0;
-  color: #ff4249;
-  margin: 0;
-  padding: 15px 15px 4px;
-`;
-
-const Pre = styled.pre`
-  ${({
-    bordered,
-    theme: {
-      yoga: {
-        colors: { primary: primaryPallete, gray: grayPallete },
-      },
+import React from 'react';
+import { node, bool, oneOf } from 'prop-types';
+import CodeBlockContext from './CodeBlockContext';
+import { ReactLive, PrismHighlight, SnackEmbed } from '.';
+import { importStatement } from './shared/templates';
+import { YogaComponents, YogaIcons, NativeComponents } from './shared/modules';
+
+const CodeBlockComponent = ({ type }) =>
```

Comment:
```suggestion
const CodeBlockComponent = {
      reactLive: ReactLive,
      highlight: PrismHighlight,
      expo: SnackEmbed,
    };
    
 // using as:
 
 <CodeBlockComponent[type] />
```

What do you think?

Reply:
I like your idea of extracting very much, sir.
This implementation looks different to me though.

I loved your previous idea of making this a gatsby plugin, let's plan this!

Reply:
Loved it!
Discussing with the team, we didn't like of the `<CodeBlockComponent[type] />` directly on JSX, so We adapted, but now is so much simpler without the object literal and find.

Thanks @ggdaltoso